### PR TITLE
refactor(rpc): be more efficient when updating state with buffer

### PR DESCRIPTION
**Pull request by zizou0x**
This PR allows us to be more efficient for during buffer lookups if the protocol system is given in the request.

### CODE REVIEW

File: tycho-indexer/src/services/deltas_buffer.rs [hunk 1]
```diff
@@ -138,36 +140,64 @@ impl PendingDeltas {
...
+                PendingDeltasError::UnknownExtractor(ps.to_string())
+            })?;
+            change_found = update_state(ps, buffer, db_state)?;
+        } else {
+            // If no protocol system is provided, iterate through all buffers
+            for (name, buffer) in self.buffers.iter() {
+                change_found = update_state(name, buffer, db_state)?;
+                if change_found {
+                    break; // Stop once a change is found
+                }
```
Comment (kayibal) [inline tycho-indexer/src/services/deltas_buffer.rs]: I think we made `protocol_system` required so it should always be available. Doing it this way is very unsafe especially if you have an extractor that is syncing. I think we should error in this case.

File: tycho-indexer/src/services/deltas_buffer.rs [hunk 2]
```diff
@@ -176,27 +206,44 @@ impl PendingDeltas {
...
```
Comment (kayibal) [inline tycho-indexer/src/services/deltas_buffer.rs]: Same here, I think we should simply error if we don't get a protocol_system.

File: tycho-indexer/src/services/deltas_buffer.rs (hunk 1 above)
Comment (louise-poole) [reply tycho-indexer/src/services/deltas_buffer.rs]: I assume this is done for backwards compatibility? But I agree - I'd rather error as it's clear that not providing a protocol system has unexpected behaviour...

Comment (zizou0x) [reply tycho-indexer/src/services/deltas_buffer.rs]: addressed in https://github.com/propeller-heads/tycho-indexer/pull/408/commits/a2db521bb2a0a30f659764899c794d47c6dde6ef

File: tycho-indexer/src/services/deltas_buffer.rs (hunk 2 above)
Comment (zizou0x) [reply tycho-indexer/src/services/deltas_buffer.rs]: addressed in https://github.com/propeller-heads/tycho-indexer/pull/408/commits/a2db521bb2a0a30f659764899c794d47c6dde6ef

**Pull request merged**